backend/infrahub/graphql/mutations/repository.py

```diff
@@ -175,7 +177,9 @@ async def mutate_update(
             infrahub_branch_name=branch.name,
         )
         if context.service:
```

Comment:
We have this check in several tasks, I am not sure when `context.service` might be undefined and whether we should raise an error if it happens then.

Reply:
usually this is mainly to make mypy happy because `service` is Optional